lighthouse-core/report/formatter.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * @license
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const fs = require('fs');
+
+// Auto convert partial filenames to lookup for audits to use.
+const formatters = {};
+fs.readdirSync(__dirname + '/partials')
+  .filter(filename => filename.endsWith('.html'))
+  .forEach(filename => {
+    const baseName = filename.replace(/\.html$/, '');
+    const capsName = baseName.replace(/-/g, '_').toUpperCase();
+    formatters[capsName] = baseName;
+  });
+
+
+class Formatter {
+  /**
+   * Getter that returns a list of supported formatters, mapping an all-caps
+   * name to the basename of the available formatter partials
+   * (e.g. `CRITICAL_REQUEST_CHAINS` to `critical-request-chains).
+   * @return {!Object<string, string>}
+   */
+  static get SUPPORTED_FORMATS() {
```

Comment:
> does this need to be in caps?

no, I just left it how it was before so that audits don't have to change the `formatter: Formatter.SUPPORTED_FORMATS.USER_TIMINGS` (or whatever) line in `extendedInfo`.

Selecting the partial to use is a little awkward in general. It could also have been a string (like e.g `requiredArtifacts` is) since that's all that variable really is, but there is the advantage of if your editor does support code completion you get suggestions there. The `SUPPORTED_FORMATS` could have lived somewhere else, too (like on `ReportGenerator`).

Basically I didn't see an option that was legitimately better, so I just left it the same until we do come up with something :)